inc/mailcollector.class.php

```diff
@@ -1354,79 +1355,71 @@ function getAdditionnalHeaders(\Laminas\Mail\Storage\Message $message) {
    **/
    function getHeaders(\Laminas\Mail\Storage\Message $message) {
 
-      $h_sender = $message->getHeader('from')->getAddressList();
-      $sender = $h_sender->current();
+      $sender_email = $this->getEmailFromHeader($message, 'from');
 
-      $h_to = $message->getHeader('to')->getAddressList();
-      $h_to->rewind();
-      $to = $h_to->current();
-
-      $reply_to_addr = null;
-      if (isset($message->reply_to)) {
-         $h_reply_to = $message->getHeader('reply_to')->getAddressList();
-         $reply_to   = $h_reply_to->current();
-         $reply_to_addr = Toolbox::strtolower($reply_to->getEmail());
+      if (preg_match('/^(mailer-daemon|postmaster)@/i', $sender_email) === 1) {
```

Comment:
With imap extension, we were comparing `$sender->mailbox`, which was not containing the host (i.e. `postmaster@domain.org` was returning `postmater`).
Now, we have to deal with the full email address.